buffer/src/main/java/io/netty/buffer/PoolChunk.java

```diff
@@ -182,23 +214,62 @@
         this.arena = arena;
         this.memory = memory;
         this.offset = offset;
-        memoryMap = null;
-        depthMap = null;
-        subpages = null;
-        subpageOverflowMask = 0;
         pageSize = 0;
         pageShifts = 0;
-        maxOrder = 0;
-        unusable = (byte) (maxOrder + 1);
+        runsAvailMap = null;
+        runsAvail = null;
+        subpages = null;
         chunkSize = size;
-        log2ChunkSize = log2(chunkSize);
-        maxSubpageAllocs = 0;
         cachedNioBuffers = null;
     }
 
     @SuppressWarnings("unchecked")
-    private PoolSubpage<T>[] newSubpageArray(int size) {
-        return new PoolSubpage[size];
+    private static PriorityQueue<Long>[] newRunsAvailqueueArray(int size) {
+        PriorityQueue<Long>[] queueArray = new PriorityQueue[size];
+        for (int i = 0; i < queueArray.length; i++) {
+            queueArray[i] = new PriorityQueue<Long>();
+        }
+        return queueArray;
+    }
+
+    private void insertAvailRun(int runOffset, int pages, Long handle) {
+        int pageIdxFloor = arena.pages2pageIdxFloor(pages);
+        PriorityQueue<Long> queue = runsAvail[pageIdxFloor];
+        queue.offer(handle);
```

Comment:
should we assert that `offer` always return true ?

Reply:
PriorityQueue.offer() do always return true, see javadoc. So no need to assert it.